src/rc_allocator.c

```diff
@@ -769,50 +769,61 @@ rc_allocator_assert_noleaks(rc_allocator *al)
+   platform_default_log("|%s|\n", dashes);
+   platform_default_log("| Allocator Stats                                                   |\n");
+   platform_default_log("|%s|\n", dashes);
+   // clang-format on
+
+   char   bytes_to_str[SIZE_TO_STR_LEN];
+   size_t nbytes = (al->stats.curr_allocated * al->cfg->io_cfg->extent_size);
+   size_to_fmtstr(bytes_to_str, sizeof(bytes_to_str), "(%s)", nbytes);
+
    platform_default_log(
-      "| Currently Allocated: %12lu extents (%4luGiB)          |\n",
+      "| Currently Allocated: %12lu extents %-14s          |\n",
       al->stats.curr_allocated,
-      curr_gib);
-   uint64 max_gib = al->stats.max_allocated / divider;
+      bytes_to_str);
+
+   nbytes = (al->stats.max_allocated * al->cfg->io_cfg->extent_size);
+   size_to_fmtstr(bytes_to_str, sizeof(bytes_to_str), "(%s)", nbytes);
    platform_default_log(
-      "| Max Allocated:       %12lu extents (%4luGiB)          |\n",
+      "| Max Allocated:       %12lu extents %-14s          |\n",
       al->stats.max_allocated,
-      max_gib);
-   platform_default_log(
-      "|--------------------------------------------------------------|\n");
-   platform_default_log(
-      "| Page Type | Allocations | Deallocations | Footprint (bytes)  |\n");
-   platform_default_log(
-      "|--------------------------------------------------------------|\n");
+      bytes_to_str);
+
+   // clang-format off
+   platform_default_log("|%s|\n", dashes);
+   platform_default_log("| Page Type  | Allocations | Deallocations |      Footprint         |\n");
+   platform_default_log("|            |      (Number of extents)    | # extents  (bytes)     |\n");
+   platform_default_log("|%s|\n", dashes);
+   // clang-format on
+
    int64 exp_allocated_count = 0;
    for (page_type type = PAGE_TYPE_FIRST; type < NUM_PAGE_TYPES; type++) {
-      const char *str           = page_type_str[type];
-      int64       allocs        = al->stats.extent_allocs[type];
-      int64       deallocs      = al->stats.extent_deallocs[type];
-      int64       footprint     = allocs - deallocs;
-      int64       footprint_gib = footprint / divider;
+      const char *str       = page_type_str[type];
+      int64       allocs    = al->stats.extent_allocs[type];
+      int64       deallocs  = al->stats.extent_deallocs[type];
+      int64       footprint = allocs - deallocs;
+      nbytes                = (footprint * al->cfg->io_cfg->extent_size);
+      size_to_fmtstr(bytes_to_str, sizeof(bytes_to_str), "(%s)", nbytes);
 
       exp_allocated_count += footprint;
 
-      platform_default_log("| %-10s | %11ld | %13ld | %8ld (%4ldGiB) |\n",
+      platform_default_log("| %-10s | %11ld | %13ld | %8ld %-14s|\n",
                            str,
                            allocs,
                            deallocs,
                            footprint,
-                           footprint_gib);
+                           bytes_to_str);
```

Comment:
The new output for these rows when running `./build/release/bin/unit/splinter_test test_splinter_print_diags` is shown below:

```
| Page Type  | Allocations | Deallocations |      Footprint         |
|            |      (Number of extents)    | # extents  (bytes)     |
|-------------------------------------------------------------------|
| trunk      |          10 |             0 |       10 (~1.25 MiB)   |
| branch     |       26506 |         16602 |     9904 (~1.20 GiB)   |
| memtable   |       18827 |         18432 |      395 (~49.37 MiB)  |
| filter     |        2375 |          2193 |      182 (~22.75 MiB)  |
| log        |           0 |             0 |        0 (0 bytes)     |
| superblock |           3 |             0 |        3 (384 KiB)     |
| misc       |           0 |             0 |        0 (0 bytes)     |
| lock       |           1 |             0 |        1 (128 KiB)     |
```

We get better sizes reported, showing in `MiB` and `GiB` where applicable.

Previously some of these were being reported as 0, which is slightly misleading.